src/main/ruby/core/truffle/cext.rb

```diff
@@ -124,14 +129,16 @@ def [](index)
     def []=(index, value)
       Truffle::CExt.string_pointer_write(@string, index, value)
     end
-
-    alias_method :to_str, :string
-    alias_method :to_s, :string
   end
 
-  class RStringPtrEnd < RStringPtr
+  class RStringPtrEnd < AbstractRStringPtr
+    def size
+      0
+    end
+
     def to_native
-      Rubinius::FFI::Pointer.new(Truffle::CExt.string_pointer_to_native(@string) + size)
+      Rubinius::FFI::Pointer.new(
+          Truffle::CExt.string_pointer_to_native(@string) + Truffle::CExt.string_pointer_size(@string))
```

Comment:
Should this point to the first `\0` or the last one (that is `ptr + #bytesize`) ?